src/app/users/users.service.ts

```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { DataService } from '../shared/data-service';
+import { IUser } from './user.interface';
+@Injectable({ providedIn: 'root' })
+export class UserService {
```

Comment:
why this?  remove the dataService and move the code to UserService.
Use UserService for User api

Reply:
big mistake!!! fixxed! thks